open-distro-elasticsearch-kubernetes/helm/opendistro-es/templates/elasticsearch/es-data-sts.yaml

```diff
@@ -0,0 +1,189 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  labels:
+{{ include "opendistro-es.labels.standard" . | indent 4 }}
+    role: data
+  name: {{ template "opendistro-es.fullname" . }}-data
+  namespace: {{ .Release.Namespace }}
+spec:
+  serviceName: {{ template "opendistro-es.fullname" . }}-data
+  replicas: {{ .Values.elasticsearch.data.replicas }}
+  template:
+    metadata:
+      labels:
+{{ include "opendistro-es.labels.standard" . | indent 8 }}
+        role: data
+      annotations:
+        {{/* This forces a restart if the secret config has changed */}}
+        {{- if .Values.elasticsearch.config }}
+        configchecksum: {{ include (print .Template.BasePath "/elasticsearch/es-config-secret.yaml") . | sha256sum | trunc 63 }}
+        {{- end }}
+    spec:
+{{- include "opendistro-es.imagePullSecrets" . | indent 6 }}
+      # Add toleration for not scheduling on dedicated node
+      tolerations:
```

Comment:
It's better to allow users to define their own `tolerations` in order to execute ES in a dedicated set of nodes. It also requires `nodeSelector` or another way to define in which nodes should pods run (such as `affinity`).